src/main/java/org/broadinstitute/consent/http/resources/DarCollectionResource.java

```diff
@@ -77,6 +80,27 @@ public Response getCollectionByReferenceId(
     }
   }
 
+  @GET
+  @Path("query")
+  @Produces("application/json")
+  @RolesAllowed(RESEARCHER)
+  public Response getCollectionsByInitialQuery(
+    @Auth AuthUser authUser,
+    @QueryParam("filterTerm") String filterTerm,
+    @QueryParam("sortField") String sortField,
+    @QueryParam("sortOrder") String sortOrder,
+    @QueryParam("pageSize") int pageSize
+  ) {
+    try {
+      User user = userService.findUserByEmail(authUser.getEmail());
+      PaginationToken token = new PaginationToken(1, pageSize, sortField, sortOrder, filterTerm, DarCollection.acceptableSortFields);
```

Comment:
Related to my comment above, maybe this is a good place to validate the provided fields (and others as well)? For example, if I called this endpoint with a `sortField` of `electionid`, that might trigger a bad request.

Reply:
That's what was done initially, these updates were added based on feedback
https://github.com/DataBiosphere/consent/pull/1273#discussion_r715055899

Reply:
The feedback in [#1273](https://github.com/DataBiosphere/consent/pull/1273#discussion_r715055899) was to move the fields themselves, which was a good change that you made. My feedback here is of a different category. The definition of the sort fields is one thing, and then what we do with those fields is another thing. My feedback here is about how we use those fields and validating that the information used to create the token is correct or not.

Reply:
That's the purpose of acceptableSortFields in the PaginationToken itself which would throw errors initially as seen here (see constructor and helper functions)

https://github.com/DataBiosphere/consent/blob/ce2a7422daf85592c8892f80b7ed3fe8af2c9074/src/main/java/org/broadinstitute/consent/http/models/PaginationToken.java

I updated that based on this comment https://github.com/DataBiosphere/consent/pull/1273#discussion_r715057680, but I'm guessing you meant for that to be applied to `null` arguments? If so, I can bring back the old implementation.

Reply:
My only real concern is that we are using `dar_code` inside of the `PaginationToken` class which doesn't separate concerns. The token class shouldn't be defaulting to a specific model class just because it's the first one we're implementing. My suggestion here in the resource class was just one way to potentially solve that problem. I'm open to any solution so long as we can fail fast in the resource class and separate `PaginationToken` from the model.